src/compiler/types.ts

```diff
@@ -1677,7 +1677,6 @@ export interface Identifier extends PrimaryExpression, Declaration, JSDocContain
      * Text of identifier, but if the identifier begins with two underscores, this will begin with three.
      */
     readonly escapedText: __String;
-    readonly originalKeywordKind?: SyntaxKind;                // Original syntaxKind which get set so that we can report an error later

```

Comment:
My only concern with this is that it is a public API breaking change. Do we expose any other way to get this information? `stringToToken` is currently marked `/** @internal */`.

If we find that there are consumers in the field that need this information, we could potentially expose it via a getter in the `src/deprecatedCompat` project with a `@deprecated` comment and the caveat that repeated access would be slower since it would be recalculated each time.

Reply:
We could try to keep it as a public but deprecated property. I do see some usage in

https://github.com/search?q=originalKeywordKind+NOT+path%3A%2F%5Elib%5C%2F%2F+NOT+path%3A%2F%5Ebin%5C%2F%2F+NOT+path%3A%2F%5E%5C%2Flib%5C%2F%2F+NOT+path%3A%2Fsrc%5C%2Fcompiler%5C%2F%2F+NOT+path%3A%2Fsrc%5C%2Fharness%5C%2F%2F+NOT+path%3A%2Fsrc%5C%2Frules%5C%2F%2F&type=code

Reply:
Is there a best practice to add deprecated accessors to prototypes right now?

Reply:
Also https://sourcegraph.com/search?q=context:global+originalKeywordKind+-repo:microsoft/TypeScript+-file:%5C.d%5C.ts+lang:typescript+-file:src/compiler+-file:src/services&patternType=standard&sm=1 which sorts a little better; it is being used by eslint, angular, a few other big projects.

Reply:
There's no prototype to add it to, since everything inherits from `Node`. You would have to add it to the instance. There's an `addNodeFactoryPatcher` function that lets you hook into the creation of a `NodeFactory` to patch in a deprecation, but it doesn't affect factories that already exist (i.e., `parseNodeFactory` in parser.ts, the internal `factory` inside of the `Parser` namespace in parser.ts, or `factory` in nodeFactory.ts).

The current patching mechanism was primarily to handle patching the factory API itself, and only the one that is reachable during transformations. As a result, the current patching mechanism doesn't touch the factory used internally by the parser, which is the one you'd want most if you want to patch node instances returned by the API.

Reply:
It seems like the benefit of the PR is somewhere in the noise, and given some of the big names who use it, they'll get slower via the accessors (though they can be cached); is the PR in general worth the break?

Reply:
It does seem like a win in terms of memory usage. If we really want to avoid the break, https://github.com/microsoft/TypeScript/pull/51498 does that.

Reply:
Ah, I didn't notice the memory improvement.

Reply:
Actually, I misspoke. we could potentially add it to the prototype of the `Identifier` constructor we return from `objectAllocator`, although this is overridden in services so you'd need to do that in two places.

Reply:
Yeah, honestly, maybe it won't be that slow and sticking it on Identifier is enough. (I was hoping caching it somehow would help but that might be slower in general...)